test/linux/example/media_on_linux/SocketCANInterface.hpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ */
+
+#ifndef LIBUAVCAN_EXAMPLE_SOCKETCANINTERFACE_HPP_INCLUDED
+#define LIBUAVCAN_EXAMPLE_SOCKETCANINTERFACE_HPP_INCLUDED
+
+#include <queue>
+#include <memory>
+
+#include <linux/can.h>
+#include <linux/can/raw.h>
+
+#include <sys/socket.h>
+
+#include "libuavcan/libuavcan.hpp"
+#include "libuavcan/media/interfaces.hpp"
+#include "libuavcan/media/can.hpp"
+
+namespace libuavcan
+{
+/**
+ * @defgroup examples Examples
+ *
+ * @{
+ * @file
+ * @namespace example   Namespace containing example implementations of libuavcan.
+ */
+namespace example
+{
+using CANFrame = libuavcan::media::CAN::Frame<libuavcan::media::CAN::TypeFD::MaxFrameSizeBytes>;
+using SocketCANFrame                = ::canfd_frame;
+static constexpr size_t ControlSize = sizeof(cmsghdr) + sizeof(::timeval);
+using ControlStorage                = typename std::aligned_storage<ControlSize>::type;
+
+/**
+ * Example of a media::Interface implemented for <a
+ * href="https://www.kernel.org/doc/Documentation/networking/can.txt">SocketCAN</a>.
+ */
+class SocketCANInterface : public libuavcan::media::Interface<CANFrame, 4, 4>
+{
+public:
+    struct Statistics
+    {
+        std::uint32_t rx_total   = 0;
```

Comment:
At a modest rate of 1k frames per second this will overflow in ~49 days. Overflowing counters are unsafe; why not use 64-bit values?

Reply:
sure. why not.